prow/plank/controller.go

```diff
@@ -236,10 +236,17 @@ func (c *Controller) syncPendingJob(pj kube.ProwJob, pm map[string]kube.Pod, rep
 		// deleted manually. Start a new pod.
 		id, pn, err := c.startPod(pj)
 		if err != nil {
-			return fmt.Errorf("error starting pod: %v", err)
+			_, isUnprocessable := err.(kube.UnprocessableEntityError)
+			if !isUnprocessable {
+				return fmt.Errorf("error starting pod: %v", err)
+			}
+			pj.Status.State = kube.ErrorState
+			pj.Status.CompletionTime = time.Now()
+			pj.Status.Description = "Job cannot be processed."
```

Comment:
We should include the error reason as well. The API server gives a handy message like "job name blabla-123.5 is not a valid DNS name" or something that we should see.

Reply:
Thought about it, just not sure how will it show up on Github. I feel though that the reports package is not meant to handle reports only to Github.

Reply:
We can log it as a warning to start out with. Maybe it's worth putting a "reason" string in the ProwJob struct that can be exposed by deck, but for now I'm happy with just logging it.

Reply:
Logging SGTM. Updated.